### PR TITLE
Update Kubernetes to v1.22.0

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

* Update Kubernetes to v1.22.1
* Update dependent modules to track with upstream

#### Verification ####

`kubectl version`

#### Linked Issues ####

For rancher/rke2#1026

#### User-Facing Change ####
```release-note
Update Kubernetes to v1.22.0
```

### CODE REVIEW

**Comment by brandond** [comment]
Currently waiting on merge of https://github.com/traefik/traefik-helm-chart/pull/451 and release of Traefik 2.5 with support for ingress v1; 2.4.x still looks for ingress v1beta1

**Review by galal-hussein — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Comment by brandond** [comment]
Got a clean CI run, just need a LGTM to push tags to k3s-io/kubernetes and k3s-io/containerd

**Review by briandowns — APPROVED** [review]

**Review by erikwilson — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
Seems to caused regression https://github.com/k3s-io/k3s/issues/3900

**Pull request merged**
